index.jsx

```diff
@@ -141,6 +141,10 @@ export default {
                 dispatch(jlinkTargetActions.recover());
                 break;
             }
+            case targetActions.RECOVER_AND_WRITE_START: {
```

Comment:
I suggest to look into a solution where actions dispatched by the app itself don't need to be intercepted in the middleware.

Reply:
Agree, I consider to move all similar actions to the right place.

Reply:
Updated.